ensime_shared/ensime.py

```diff
@@ -136,16 +136,23 @@ def queue_poll(self, sleep_t=0.5):
 
         Value of sleep is low to improve responsiveness.
         """
+        connection_alive = True
         while self.running:
             if self.ws:
                 def logger_and_close(m):
                     self.log("Websocket exception: {}".format(m))
-                    self.teardown()
+                    if not self.running:
```

Comment:
Can this condition ever be hit? on the first exception, `self.running= True`, which means we call `teardown()`, which flips `self.running= False`. This is all during the same pass. On the next pass, the `self.running` condition will evaluate to `False`, terminating the loop, which prevents any future evaluations of `logger_and_close()`.